controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java

```diff
@@ -85,20 +85,27 @@ public boolean authenticate(Map<String, String> headers) {
     }
 
     @Override
-    public Permissions authorize(String resource, Map<String, String> headers) {
-        String userName = headers.get("username");
+    public Permissions authorize(String resource, String token) {
+        String[] parts = parseToken(token);
+        String userName = parts[0];
+
         if (Strings.isNullOrEmpty(userName) || !userMap.containsKey(userName)) {
             throw new CompletionException(new AuthenticationException(userName));
         }
         return authorizeForUser(userMap.get(userName), resource);
-
     }
 
     @Override
     public void initialize(ServerConfig serverConfig) {
         loadPasswordFile(((GRPCServerConfig) serverConfig).getUserPasswordFile());
     }
 
+    private static String[] parseToken(String token) {
+        String[] parts = new String(Base64.getDecoder().decode(token), Charsets.UTF_8).split(":", 2);
+        Preconditions.checkArgument(parts.length == 2, "Invalid authorization token");
```

Comment:
OK, here is the validation that was talking about above. Is checking the length sufficient? But, this is validating only when parsing, so I'm wondering whether we should be validating this earlier.